src/Select.js

```diff
@@ -177,7 +177,7 @@ var Select = React.createClass({
 					newProps.placeholder)
 				);
 			};
-			if (this.props.asyncOptions) {
+			if (this.props.asyncOptions && this.props.autoload) {
```

Comment:
I agree with @TigerC10. It's something I specifically need, another use case to consider is in poor network situations. What happens right now is that the react-select will requests options instantly, but no value will be displayed, as long as the call is not done. Which is obviously not satisfactory.